From b22e69f0901f28e12027936d914ffc307d637677 Mon Sep 17 00:00:00 2001
From: Jakub Dorzak <jdorzakx@intel.com>
Date: Tue, 26 Jul 2016 14:59:58 +0100
Subject: [PATCH 046/102] ASoC: Intel: Boards: Added option to enable TDF8532
 DAI link

Added Kconfig option that allows user to switch between TDF8532
and snd-soc-dummy in DAI link struct for GPMRB machine driver.

Signed-off-by: Jakub Dorzak <jdorzakx@intel.com>

Conflicts:
	sound/soc/intel/Kconfig

Conflicts:
	sound/soc/intel/Kconfig
---
 sound/soc/intel/boards/Kconfig     | 11 +++++++++++
 sound/soc/intel/boards/bxt_gpmrb.c |  5 +++++
 2 files changed, 16 insertions(+)

diff --git a/sound/soc/intel/boards/Kconfig b/sound/soc/intel/boards/Kconfig
index df68be6..3655cac 100644
--- a/sound/soc/intel/boards/Kconfig
+++ b/sound/soc/intel/boards/Kconfig
@@ -408,12 +408,23 @@ config SND_SOC_INTEL_BXT_GPMRB_MACH
 	select SND_SOC_INTEL_SST
 	select SND_SOC_TDF8532
 	select SND_SOC_COMPRESS
+	select DW_DMAC
 	default n
 	help
 	   This adds support for ASoC machine driver for Broxton IVI GP MRB platform
 	   Say Y if you have such a device
 	   If unsure select "N".
 
+config SND_SOC_INTEL_BXT_GPMRB_MACH_TDF8532_CODEC
+	bool "Enable TDF8532 codec"
+	depends on SND_SOC_INTEL_BXT_GPMRB_MACH
+	select SND_SOC_TDF8532
+	default y
+	help
+	   This adds support for TDF8532 codec to BE DAI link.
+	   Say "Y" if you are using GP MRB board.
+	   Otherwise choose "N".
+
 config SND_SOC_INTEL_APLI_LHCRB_DUMMY_MACH
 	tristate "ASoC Audio driver for APLI LHCRB Dummy Codec"
 	depends on X86_INTEL_LPSS && I2C
diff --git a/sound/soc/intel/boards/bxt_gpmrb.c b/sound/soc/intel/boards/bxt_gpmrb.c
index 3948257..2946b55 100644
--- a/sound/soc/intel/boards/bxt_gpmrb.c
+++ b/sound/soc/intel/boards/bxt_gpmrb.c
@@ -266,8 +266,13 @@
 		.name = "SSP4-Codec",
 		.be_id = 0,
 		.cpu_dai_name = "SSP4 Pin",
+#ifdef CONFIG_SND_SOC_INTEL_BXT_GPMRB_MACH_TDF8532_CODEC
 		.codec_name = "i2c-INT34C3:00",
 		.codec_dai_name = "tdf8532-hifi",
+#else
+		.codec_name = "snd-soc-dummy",
+		.codec_dai_name = "snd-soc-dummy-dai",
+#endif
 		.platform_name = "0000:00:0e.0",
 		.ignore_suspend = 1,
 		.dpcm_playback = 1,
-- 
1.9.1

